File: backend/src/main/java/com/svelteup/app/backend/profile/dtos/SvelteUpUserAccountDto.java
package com.svelteup.app.backend.profile.dtos;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.ArrayList;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class SvelteUpUserAccountDto {
   public Integer phoneNumberAreaCode;
   public Integer phoneNumberPhoneNumber;

   public String firstName;
   public String lastName;
   public String email;

   public String addressLineOne;
   public String userProfileCity;
   public String userProfileState;
   public String userProfileCountry;
   public Integer zipCode;

   public String aboutMe;
   public String image;

   List<String> foeList = new ArrayList<>();
}

File: frontend/src/Dto/profile/SvelteUpUserAccountDto.ts
class SvelteUpUserAccountDto
{
    public  phoneNumberAreaCode:number|null;
    public  phoneNumberPhoneNumber:number|null

    public  firstName:string|null;
    public  lastName:string|null;
    public  email:string|null;

    public addressLineOne:string|null;
    public userProfileCity:string|null;
    public userProfileState:string|null;
    public userProfileCountry:string|null;
    public zipCode:number|null;

    public aboutMe:string|null;
    public image:object|null;

    public constructor() {
    }

    public buildEmailDto(email:string)
    {
        this.email = email;
    }

    public buildProfileDto(image:any, aboutMe:string)
    {
        this.image = image;
        this.aboutMe = aboutMe;
    }

    public buildIdentityDto(firstName:string, lastName:string, areaCode:number,phoneNumberNumber:number, addressLine:string, userProfileState:string, userProfileCountry:string)
    {
        this.firstName  = firstName;
        this.lastName = lastName;
        this.phoneNumberAreaCode = areaCode;
        this.addressLineOne  = addressLine;
        this.userProfileState  = userProfileState;
        this.userProfileCountry  = userProfileCountry;
    }
}

export default SvelteUpUserAccountDto;